docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -635,310 +960,179 @@ public class CustomSecurityIdentityAugmentor implements SecurityIdentityAugmento
-quarkus.oidc.credentials.secret=mysecret
-----
-
-or
-
-[source,properties]
-----
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
-quarkus.oidc.credentials.client-secret.value=mysecret
+}
 ----
 
-or with the secret retrieved from a xref:credentials-provider.adoc[CredentialsProvider]:
+Now, the following code will work when the user is signing in into your application with both Google or GitHub:
 
-[source,properties]
+[source,java]
 ----
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
 
-# This is a key which will be used to retrieve a secret from the map of credentials returned from CredentialsProvider
-quarkus.oidc.credentials.client-secret.provider.key=mysecret-key
-# Set it only if more than one CredentialsProvider can be registered
-quarkus.oidc.credentials.client-secret.provider.name=oidc-credentials-provider
-----
+import io.quarkus.security.Authenticated;
+import io.quarkus.security.identity.SecurityIdentity;
 
-`client_secret_post`:
+@Path("/service")
+@Authenticated
+public class TokenResource {
 
-[source,properties]
-----
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
-quarkus.oidc.credentials.client-secret.value=mysecret
-quarkus.oidc.credentials.client-secret.method=post
-----
+    @Inject
+    SecurityIdentity identity;
 
-`client_secret_jwt`, signature algorithm is HS256:
+    @GET
+    @Path("/google")
+    @Produces("application/json")
+    public String getUserName() {
+        return identity.getPrincipal().getName();
+    }
 
-[source,properties]
-----
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
-quarkus.oidc.credentials.jwt.secret=AyM1SysPpbyDfgZld3umj1qzKObwVMkoqQ-EstJQLr_T-1qS0gZH75aKtMN3Yj0iPS4hcgUuTwjAzZr1Z9CAow
+    @GET
+    @Path("/github")
+    @Produces("application/json")
+    public String getUserName() {
+        return identity.getPrincipal().getUserName();
+    }
+}
 ----
 
-or with the secret retrieved from a xref:credentials-provider.adoc[CredentialsProvider]:
+Possibly a simpler alternative is to inject both `@IdToken JsonWebToken` and `UserInfo` and use `JsonWebToken` when dealing with the providers returning `IdToken` and `UserInfo` - with the providers which do not return `IdToken`.
 
-[source,properties]
-----
-quarkus.oidc.auth-server-url=http://localhost:8180/realms/quarkus/
-quarkus.oidc.client-id=quarkus-app
+The last important point is to make sure the callback path you enter in the GitHub OAuth application configuration matches the endpoint path where you'd like the user be redirected to after a successful GitHub authentication and application authorization, in this case it has to be set to `http:localhost:8080/github/userinfo`.
```

Comment:
```suggestion
The last important point is to make sure the callback path you enter in the GitHub OAuth application configuration matches the endpoint path where you want to redirect the user after a successful GitHub authentication and application authorization.
In this case, you must set the callback path to `http:localhost:8080/github/userinfo`.
```

Reply:
Suggestion: I would also suggest removing "important point" here because it might imply that the other point are not/less important. What about rephrasing to "You must ensure that the callback path you enter in the GitHub....." and if it is of particular importance to highlght, this section could be encoded in an [IMPORTANT] block?

Reply:
@sheilamjones Sure, can you please propose a change I can commit ?

Reply:
@sheilamjones Updated as you proposed